js/src/components/app-select-control/index.js

```diff
@@ -16,14 +16,34 @@ import './index.scss';
  * it will be added to the container div's `className`,
  * so that you can further control its style.
  *
- * @param {*} props
+ * @param {*} props The component props.
+ * @param {boolean} [props.selectSingleValue=false] Whether the select should show only one value.
  */
 const AppSelectControl = ( props ) => {
-	const { className, ...rest } = props;
+	const { className, options, selectSingleValue = false, ...rest } = props;
+	const showSingleValue = selectSingleValue && options?.length === 1;
+
+	let selectProps = {
+		options,
+		...rest,
+	};
+
+	if ( showSingleValue ) {
+		selectProps = {
+			...selectProps,
+			suffix: ' ',
+			tabIndex: '-1',
+			readOnly: true,
+		};
+	}
 
 	return (
-		<div className={ classNames( 'app-select-control', className ) }>
-			<SelectControl { ...rest } />
+		<div
+			className={ classNames( 'app-select-control', className, {
+				'app-select-control--has-single-value': showSingleValue,
```

Comment:
```suggestion
				'gla_app-select-control--single-value-readonly': showSingleValue,
```

Can we have more descriptive class name like this?

Reply:
@ankitrox See my comment below for the prefix to be used. Also adding `readonly` to the class name does not have any semantic meaning. 👍